### PR TITLE
LINK 1075, part 1 | Create signup group API endpoint

### PR DESCRIPTION
**Pull request by harriris-vincit**
### Description
Adds a new API endpoint for creating and retrieving the details of `SignUpGroup` instances. `SignUp` instances belonging to the same group are also created at the same time as the group through this new endpoint.

The rest of the signup group functionality, updating and deleting the groups, are going to be included in separate PRs. Since the group enrolment DB structure is not yet in the main branch, this PR uses the DB structure feature branch (LINK-1294) as the target branch.
### Partially closes
[LINK-1075](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1075)

[LINK-1075]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1075?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: registrations/serializers.py
Comment (jorilindell) [inline registrations/serializers.py:353]: Now `POST /signup_group/` response contains "attending" and "waitlisted" fields, but also "signups" field. I think "attending" and "waitlist" just have duplicate information comparing to "signups" and can removed totally.

Example response at the moment:
```
{
    "id": 2,
    "registration": 2,
    "signups": [
        {
            "id": 102,
            "service_language": null,
            "created_at": "2023-08-28T11:53:32.566984Z",
            "last_modified_at": "2023-08-28T11:53:32.567080Z",
            "created_by": "Jori Lindell - jori.lindell@gmail.com",
            "last_modified_by": "Jori Lindell - jori.lindell@gmail.com",
            "responsible_for_group": true,
            "first_name": "Jori",
            "last_name": "Lindell 1",
            "date_of_birth": "2011-01-01",
            "city": null,
            "email": "jori.lindell@gmail.com",
            "extra_info": null,
            "membership_number": null,
            "phone_number": null,
            "notifications": "none",
            "attendee_status": "attending",
            "street_address": null,
            "zipcode": "00100",
            "presence_status": "not_present",
            "registration": 2,
            "signup_group": 2,
            "native_language": null
        }
    ],
    "extra_info": null,
    "attending": {
        "count": 1,
        "people": [
            {
                "id": 102,
                "service_language": null,
                "created_at": "2023-08-28T11:53:32.566984Z",
                "last_modified_at": "2023-08-28T11:53:32.567080Z",
                "created_by": "Jori Lindell - jori.lindell@gmail.com",
                "last_modified_by": "Jori Lindell - jori.lindell@gmail.com",
                "responsible_for_group": true,
                "first_name": "Jori",
                "last_name": "Lindell 1",
                "date_of_birth": "2011-01-01",
                "city": null,
                "email": "jori.lindell@gmail.com",
                "extra_info": null,
                "membership_number": null,
                "phone_number": null,
                "notifications": "none",
                "attendee_status": "attending",
                "street_address": null,
                "zipcode": "00100",
                "presence_status": "not_present",
                "registration": 2,
                "signup_group": 2,
                "native_language": null
            }
        ]
    },
    "waitlisted": {
        "count": 0,
        "people": []
    }
}
```

Comment (harriris-vincit) [reply registrations/serializers.py]: @jorilindell I've now removed the two fields as suggested in https://github.com/City-of-Helsinki/linkedevents/pull/675/commits/ac5875086bf0548b7efa7cea3deb5437b66a97c9.

File: registrations/tests/test_signup_group_get.py
Comment (voneiden) [inline registrations/tests/test_signup_group_get.py:76]: Trivial: not a fan of the double underscore and mixing both styles (`test_x_y_z` vs `test__x_y_z` inside a single file), but on the other hand it seems to be consistent with the codebase..

**Review by voneiden — APPROVED** [review]
Functionally LGTM, there's some duplication in the tests, but lets see what sonarcloud  thinks about it.

**Review by voneiden — CHANGES_REQUESTED** [review]
Ok, pointing now towards master. Let's get the necessary fixes for tests in. At least one of the failures, if not all, relate to this PR: https://github.com/City-of-Helsinki/linkedevents/pull/676

File: registrations/serializers.py
Comment (voneiden) [inline registrations/serializers.py:365]: @tuomas777 raised some concerns on the use of `__all__` without having a test checking which fields are returned by the API. This is to avoid accidental leaks of personal information, if new fields are added later (been there done that ). 

Tuomas requests a test case to verify the returned fields (as opposed to just enumerating the fields with `fields`)

Comment (voneiden) [reply registrations/serializers.py:365]: See utility method 'assert_fields_exist`

Comment (harriris-vincit) [reply registrations/serializers.py:365]: Yes, from a security standpoint not only testing for, but also explicitly defining only the fields that can be shown makes a lot of sense. I'll do this change in a moment.

Comment (harriris-vincit) [reply registrations/serializers.py]: @voneiden now changed in https://github.com/City-of-Helsinki/linkedevents/pull/675/commits/ec65bac9ed6106573548849220928f38d6c37d33

**Review by jorilindell — COMMENTED** [review]
`test_return_correct_routes` unit test should be fixed. `signup_group` endpoints is now added to API documentation page so the test fails. Assertion for `signup_group` route could be also added. e.g.
```
assert (response.data["signup_group"] == "http://testserver/v1/signup_group/")
```

**Review by jorilindell — APPROVED** [review]
LGTM

File: registrations/api.py
Comment (harriris-vincit) [inline registrations/api.py:372]: @jorilindell & @voneiden: I changed the permissions to be the same for signup groups and signups (something that wasn't noticed earlier). Perhaps check this once more for just the permissions and approve if OK

**Review by voneiden — APPROVED** [review]
ok!

**Pull request merged**
